hwsecurity/fido2/src/main/java/de/cotech/hw/fido2/internal/ctap2/Ctap2CborSerializer.java

```diff
@@ -265,4 +285,53 @@ private void writeToMap(MapBuilder<?> mapBuilder, PublicKeyCredentialDescriptor
             }
         }
     }
+
+    private void writeToMap(MapBuilder<?> mapBuilder, ExtensionParameter extensionParameters) {
+        Object value = extensionParameters.value();
+        if (value instanceof BooleanExtensionParameterValue) {
+            mapBuilder.put(extensionParameters.key(), ((BooleanExtensionParameterValue) value).value());
+        } else if (value instanceof ByteArrayExtensionParameterValue) {
+            mapBuilder.put(extensionParameters.key(), ((ByteArrayExtensionParameterValue) value).value());
+        } else {
+            throw new UnsupportedOperationException();
+        }
+    }
+
+    private void writeToMap(MapBuilder<?> mapBuilder, ExtensionParameter extensionParameters, AuthenticatorGetAssertion aga) {
+        // Handle parameters which require assertion-specific data here
+        Object value = extensionParameters.value();
+        if (value instanceof HmacSecretExtensionParameterValue) {
+            HmacSecretExtensionParameterValue param = (HmacSecretExtensionParameterValue) value;
```

Comment:
Significant change here: serialize hmac-secret extension parameters to the CTAP2 CBOR request.